android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -300,6 +303,40 @@ fun Resource.appendPractitionerInfo(practitionerId: String?) {
   }
 }
 
+fun Resource.appendRelatedEntityLocation(
+  questionnaireResponse: QuestionnaireResponse,
+  questionnaireConfig: QuestionnaireConfig,
+  context: Context,
+) {
+  val locationCoding =
+    Coding().apply {
+      system =
+        context.getString(
+          org.smartregister.fhircore.engine.R.string.sync_strategy_related_entity_location_system,
+        )
+      display =
+        context.getString(
+          org.smartregister.fhircore.engine.R.string.sync_strategy_related_entity_location_display,
+        )
+    }
+  questionnaireConfig.linkIds
+    ?.filter { it.type == LinkIdType.LOCATION }
+    ?.forEach { linkIdConfig ->
+      val locationId =
+        questionnaireResponse
+          .find(linkIdConfig.linkId)
+          ?.itemFirstRep
+          ?.answerFirstRep
+          ?.valueStringType
+          ?.value
+
+      val existingTag = this.meta.getTag(locationCoding.system, locationId)
+      if (!locationId.isNullOrEmpty() && existingTag == null) {
+        this.meta.addTag(locationCoding.apply { setCode(locationId) })
+      }
```

Comment:
this seems like the cheaper operation (is that correct?), I'd move it above the locationID and only get locationID if this as null, don't even need to assign this, eg s/t like
```suggestion
      if(this.meta.getTag(locationCoding.system, locationId) == null) {
          val locationId =
            questionnaireResponse
              .find(linkIdConfig.linkId)
              ?.itemFirstRep
              ?.answerFirstRep
              ?.valueStringType
              ?.value
    
          if (!locationId.isNullOrEmpty()) {
            this.meta.addTag(locationCoding.apply { setCode(locationId) })
      }
```

Reply:
Retrieving the `locationId` can be expensive depending on the number of items to be iterated. This can be done outside this function and the id passed as a parameter.

Reply:
I see, makes sense, cool